### PR TITLE
Feature/query cxx yan sfcshp prepbufr

### PR DESCRIPTION
**Pull request by YanHao-NOAA**
## Description
Added yaml file for sfcshp_prepbufr, and testinput/output files.

### Issue(s) addressed

Link the issues to be closed with this PR
- fixes #<issue_number>

If this does not close an existing issue, then [be sure to add a Pipeline, Label, Estimate, Assignees, and Epic](https://jointcenterforsatellitedataassimilation-jedi-docs.readthedocs-hosted.com/en/latest/inside/practices/issues.html)

## Acceptance Criteria (Definition of Done)

What does it mean for this PR to be finished?

## Dependencies

If there are PRs that need to be merged before or along with this one, please add "waiting for another PR" label and list the dependencies in the other repositories (example below). Note that the branches in the other repositories should have matching names for the automated tests to pass.
Waiting on the following PRs:
- [ ] waiting on JCSDA/eckit/pull/<pr_number>
- [ ] waiting on JCSDA/atlas/pull/<pr_number>

## Impact

Please list the other repositories (if any) that this PR will require changes in (example below)

- [ ] saber
- [ ] ufo
- [ ] ...

### CODE REVIEW

**Comment by YoulongXia-NOAA** [comment]
@YanHao-NOAA, I just reviewed your PR today, I also add @emilyhcliu  and @rmclaren  there. I am not sure why one wrong person's name is added. I am trying to remove that name, I cannot do that. Nevertheless,  please look at my comments and hopefully they can help you to polish your yaml file.

**Comment by YoulongXia-NOAA** [comment]
@YanHao-NOAA, I remove the wrong person now. I found many variables with definitions in Greg Thomson's table and  ObsSpace.yaml. You may need to ask Greg to add. First please check the table and see if a definition can be used (https://docs.google.com/spreadsheets/d/1fHb2QIHOwgp4d6plyn9Dj9HZFZsDRxcPaUV6hdTb5BU/edit#gid=0).

**Comment by emilyhcliu** [comment]
@YanHao-NOAA You have 21 files modified in this PR compared to the base feature/query_cxx.

**Comment by emilyhcliu** [comment]
> @YanHao-NOAA You have 21 files modified in this PR compared to the base feature/query_cxx.

I think this is probably because you merger develop into your branch.  Your branch is branched off from feature/query_cxx so so you would want to merge feature/query_cxx into your branch.

**Comment by emilyhcliu** [comment]
@YanHao-NOAA How many files you added/modified in your branch for sfcshp data?
Was it three files?
(1) ./test/testinput/bufr_ncep_prepbufr_sfcshp.yaml
(2)./test/testinput/SFCSHP.prepbufr
(3).test/testout/bufr_ncep_prepbufr_sfcshp.nc

**Comment by emilyhcliu** [comment]
@YanHao-NOAA I figured out what was the problem.  I merged develop into feature/query_cxx and then update your branch with feature/qudry_cxx.  Then the files modified were down to 3 from 21.

File: test/testinput/bufr_ncep_sfcshp.yaml [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+          cloudType:
+            query: "*/CLDSQ1/CLTP"
+
+          heightOfBaseOfCloud:
+            query: "*/CLDSQ1/HOCB"
+
+          cloudCoverTotal:
+            query: "*/TOCC"
+
```
Comment (YoulongXia-NOAA) [inline test/testinput/bufr_ncep_sfcshp.yaml:86]: cloudCoverTotal:
   query: "*/TOCC"
   transforms:
     - scale: 0.01

File: test/testinput/bufr_ncep_sfcshp.yaml [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+          longName: "Station Pressure"
+          units: "Pa"
+          range: [90000, 105000]
+
+        - name: "ObsValue/presentWeather"
+          coordinates: "longitude latitude"
+          source: variables/presentWeather
+          longName: "Present Weather"
+          units: ""
+
```
Comment (YoulongXia-NOAA) [inline test/testinput/bufr_ncep_sfcshp.yaml:146]: Remove  units: ""

File: test/testinput/bufr_ncep_sfcshp.yaml [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+          longName: "Dewpoint Temperature"
+          units: "K"
+          range: [200, 315]
+
+        - name: "MetaData/waterTemperatureMethod"
+          coordinates: "longitude latitude"
+          source: variables/waterTemperatureMethod
+          longName: "Method of Sea Surface Measurement"
+          units: ""
+
```
Comment (YoulongXia-NOAA) [inline test/testinput/bufr_ncep_sfcshp.yaml:165]: Remove  units: ""

File: test/testinput/bufr_ncep_sfcshp.yaml [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+          units: "K"
+          range: [260, 315]
+
+        - name: "ObsValue/seaSurfaceSalinity"
+          coordinates: "longitude latitude"
+          source: variables/sss
+          longName: "Sea Surface Salinity"
+          units: "parts/1000"
+          range: [20, 50]
+
```
Comment (YoulongXia-NOAA) [inline test/testinput/bufr_ncep_sfcshp.yaml:179]: I did not find seaSurfaceTemperature in convention table, please add it in ObsSpace.yaml. 
units: "parts/1000" => g/kg. If the units is not g/kg, you will need to use transforms to convert.

File: test/testinput/bufr_ncep_sfcshp.yaml [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+          source: variables/depthBelowWaterSurface
+          longName: "Depth Below Water Surface"
+          units: "m"
+#         range: [-2000, 0]
+
+        - name: "QualityMarker/verticalSignificanceSurfaceObservations"
+          coordinates: "longitude latitude"
+          source: variables/verticalSignificanceSurfaceObservations
+          longName: "Vertical Significance (Surface Observations)"
+          units: ""
```
Comment (YoulongXia-NOAA) [inline test/testinput/bufr_ncep_sfcshp.yaml:191]: Remove units: ""

File: test/testinput/bufr_ncep_sfcshp.yaml [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+          source: variables/cloudAmount
+          longName: "Cloud Amount"
+          units: "1"
+
+        - name: "MetaData/cloudType"
+          coordinates: "longitude latitude"
+          source: variables/cloudType
+          longName: "Cloud Type"
+          units: ""
+
```
Comment (YoulongXia-NOAA) [inline test/testinput/bufr_ncep_sfcshp.yaml:204]: The same as above

File: test/testinput/bufr_ncep_sfcshp.yaml [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+          source: variables/cloudCoverTotal
+          longName: "Total Cloud Coverage"
+          units: "1"
+
+        - name: "ObsValue/heightAboveSurfaceofBaseofLowestCloud"
+          coordinates: "longitude latitude"
+          source: variables/heightAboveSurfaceofBaseofLowestCloud
+          longName: "Height above Surface of Base of Lowest Cloud"
+          units: ""
+
```
Comment (YoulongXia-NOAA) [inline test/testinput/bufr_ncep_sfcshp.yaml:222]: units: "" ==> units: "m". please double check if the data value is m, if not, you need to convert.

Comment (YanHao-NOAA) [reply test/testinput/bufr_ncep_sfcshp.yaml:222]: This is code table, no units

File: test/testinput/bufr_ncep_sfcshp.yaml (hunk 5 above)
Comment (YanHao-NOAA) [reply test/testinput/bufr_ncep_sfcshp.yaml:191]: done

**Comment by YoulongXia-NOAA** [comment]
@YanHao-NOAA, for your prebufr yaml file, you may need to scale your cloudCoverTotal. Please double check if the value is 0-100, if yes, you need to scale. If not, it is 0-1, please ignore my comments.

File: test/testinput/bufr_ncep_sfcshp.yaml (hunk 1 above)
Comment (YanHao-NOAA) [reply test/testinput/bufr_ncep_sfcshp.yaml:86]: Done. Thanks

**Comment by emilyhcliu** [comment]
More validation... for Sea Surface Temperature
bufr_ncep_prepbufr_sfcshp_seaSurfaceTemperature_dim0

The obs count from debufr.x is also 10079

**Comment by emilyhcliu** [comment]
Validate pressure ....
Check pressure
bufr_ncep_prepbufr_sfcshp_pressure_dim0
bufr_ncep_prepbufr_sfcshp_pressure_dim1

The obs count from debufr.x is 26352 (25239+1113)

**Comment by emilyhcliu** [comment]
@YanHao-NOAA Have you added the new variables from SFCSHP BUFR  into feature/query_cxx in IODA repository?

**Comment by emilyhcliu** [comment]
@YanHao-NOAA I changed this PR to a draft.  Let's turn it back to a PR when you are ready.

**Comment by YanHao-NOAA** [comment]
> I am working on that.

**Comment by YanHao-NOAA** [comment]
> @YanHao-NOAA Have you added the new variables from SFCSHP BUFR into feature/query_cxx in IODA repository?

@emilyhcliu  I am working on this, thanks.

**Comment by BenjaminRuston** [comment]
@emilyhcliu and @YanHao-NOAA should this be closed, and submit a new `sfc ship` as a new PR?

**Comment by BenjaminRuston** [comment]
@emilyhcliu should we close this issue, and submit a new PR to address ship obs?

**Comment by BenjaminRuston** [comment]
closing on advice of @emilyhcliu

**Pull request closed without merging**